pkg/apis/batch/validation/validation.go

```diff
@@ -485,9 +519,28 @@ func validatePodTemplateUpdate(spec, oldSpec batch.JobSpec, fldPath *field.Path,
 }
 
 // ValidateJobStatusUpdate validates an update to a JobStatus and returns an ErrorList with any errors.
-func ValidateJobStatusUpdate(status, oldStatus batch.JobStatus) field.ErrorList {
+func ValidateJobStatusUpdate(job, oldJob *batch.Job) field.ErrorList {
 	allErrs := field.ErrorList{}
-	allErrs = append(allErrs, validateJobStatus(&status, field.NewPath("status"))...)
+	statusFld := field.NewPath("status")
+	allErrs = append(allErrs, validateJobStatus(job, statusFld)...)
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.JobManagedByLabel) {
```

Comment:
Same thing here. Not clear to me why this is part of this feature and we should mention it in the error message somehow.

Reply:
Comments added, I don't think this should be mentioned in the error messages though.